### PR TITLE
JWT Vulnerability Fix - Force verification algorithm

### PR DESCRIPTION
**Pull request by KSanthanam**
This fixes the JWT vulnerability highlighted by Tim McClean in this article https://auth0.com/blog/2015/03/31/critical-vulnerabilities-in-json-web-token-libraries/

### CODE REVIEW

File: src/test/java/com/auth0/jwt/JWTVerifierTest.java [hunk 1]
```diff
@@ -25,27 +25,27 @@ public void constructorShouldFailOnEmptySecret() {
```
Comment (Rebzie) [inline src/test/java/com/auth0/jwt/JWTVerifierTest.java]: I think you should write a unit test for forcing algorithm

File: src/main/java/com/auth0/jwt/JWTVerifier.java [hunk 2]
```diff
@@ -78,9 +79,12 @@ public JWTVerifier(byte[] secret) {
      * @throws JWTVerifyException    when expiration, issuer or audience are invalid
```
Comment (Rebzie) [inline src/main/java/com/auth0/jwt/JWTVerifier.java]: update java doc

**Comment by Rebzie** [comment]
code looks good see comments on unit test and java doc

**Comment by rolodato** [comment]
Hi @KSanthanam, thanks for your pull request. However, this library isn't affected by this vulnerability because it only supports symmetric signatures: https://github.com/auth0/java-jwt/blob/master/src/main/java/com/auth0/jwt/JWTVerifier.java#L55-L58

Eventually we'd like to add support for asymmetric signatures, so we could merge this into a branch other than `master`. Could you edit this PR so that it targets the `rs` branch instead? We're not ready to start development on this, but your PR would be a good starting point. Thanks!

**Comment by pose** [comment]
@KSanthanam Also, please take into account that if you find a security issue you should follow the procedure detailed here: auth0.com/whitehat Thanks!

**Comment by KSanthanam** [comment]
@rolodato Thank you. Yes, I noticed it doesn't do RSA. Is anyone working on that? Shall I give it a crack?

**Comment by rolodato** [comment]
@KSanthanam: Not right now, but feel free to change this PR so that it targets the `rs` branch instead of `master`. That way we can use this as a base when we start development.

**Comment by silhouettes** [comment]
Two things:

1. This library now supports asymmetric signatures, and it is in master.
2. However, the vulnerability is prevented in a different way (see #53 for details).

So, it would still be nice to be able to whitelist acceptable algorithms, but this PR is significantly out of date, and not being able to do so does not seem to be critical.

**Pull request closed without merging**